File: Javascript/38. Count and Say.js
/**
 * @param {number} n
 * @return {string}
 */
var countAndSay = function(n) {
    if (n === 1) return "1";
    return rle(countAndSay(n-1))
};

var rle = function(str) {
    let count = 0
    let curr = str.charAt(0)
    let temp = ""

    for (let s = 0; s < str.length; s++) {
        if (curr === str.charAt(s)) {
            count++
        } else {
            temp += `${count}${curr}`
            count = 1
            curr = str.charAt(s)
        }
    }
    temp += `${count}${curr}`
    return temp
}

File: Python/38. Count and Say.py
class Solution:
    def countAndSay(self, n: int) -> str:     
        if n == 1:
            return "1"
        return self.rle(self.countAndSay(n - 1))
    
    def rle(self, str: str) -> str:
        count, curr = 0, str[0]
        temp = ""
        for s in str:
            if curr == s:
                count += 1
            else:
                temp += f'{count}{curr}'
                count = 1
                curr = s
        temp += f'{count}{curr}'
        
        return temp
        

if __name__ == "__main__":
    solution = Solution()
    # For testing
    # print("result:", solution.countAndSay(5))